src/encoder.rs

```diff
@@ -105,18 +123,56 @@ pub struct Encoder<W: Write> {
     info: Info,
     filter: FilterType,
     adaptive_filter: AdaptiveFilterType,
+    sep_def_img: bool,
 }
 
 impl<W: Write> Encoder<W> {
     pub fn new(w: W, width: u32, height: u32) -> Encoder<W> {
-        let mut info = Info::default();
-        info.width = width;
-        info.height = height;
+        let info = Info {
+            width,
+            height,
+            ..Default::default()
+        };
+        Encoder {
+            w,
+            info,
+            filter: FilterType::default(),
+            adaptive_filter: AdaptiveFilterType::default(),
+            sep_def_img: false,
+        }
+    }
+
+    pub fn new_animated(
+        w: W,
+        width: u32,
+        height: u32,
+        num_frames: u32,
+        num_plays: u32,
+        sep_def_img: bool,
+    ) -> Encoder<W> {
```

Comment:
This doesn't need to happen in the constructor, configuring `num_frames` etc. can be done with methods on `Encoder` similar to setting up color, compression, and default filters. I think this is a lot clearer than having 4 parameters with the same type, and a bool.